hw/hw17/html_render.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+
+"""
+Python class example.
+
+"""
+
+# The start of it all:
+# Fill it all in here.
+
+
+class Element(object):
+
+    IND_LEVEL = "    "
+
+    def __init__(self, content="", **kwargs):
+        self.children = [content] if content else []
+        self.attributes = kwargs
+
+    def append(self, new_child):
+        self.children.append(new_child)
+
+    def render(self, file_out, indent="    ", ):
+
+        if len(self.attributes) < 1:
+            file_out.write("\n%s<%s>" % (indent, self.name))
```

Comment:
You might want to instantiate variables you are using like `self.name` in your `__init__()` to `None` to avoid bugs.